hail/python/hail/linalg/blockmatrix.py

```diff
@@ -1427,6 +1459,47 @@ def __matmul__(self, b):
 
         return BlockMatrix(BlockMatrixDot(self._bmir, b._bmir))
 
+    @typecheck_method(b=oneof(np.ndarray, block_matrix_type), split_on_inner=int, path_prefix=str)
+    def tree_matmul(self, b, split_on_inner, path_prefix):
+        """Matrix multiplication in situations with large inner dimension. This function splits a single matrix
```

Comment:
We generally prefer a one sentence description on its own line, followed by a full description. Some python tools use this in tooltips